### PR TITLE
Added icon support.

### PR DESCRIPTION
**Pull request by a-hansen**
@Pacane @example6 - Icons coming in a pull request to the alarm link as an example.

### CODE REVIEW

File: sdk/dslink/src/main/java/org/dsa/iot/dslink/DSLinkHandler.java [hunk 1]
```diff
@@ -91,6 +92,43 @@ public Configuration getConfig() {
...
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try {
+            byte[] buf = new byte[512];
+            int len = in.read(buf);
+            while (len > 0) {
+                out.write(buf, 0, len);
+                len = in.read(buf);
+            }
+        } catch (Exception x) {
+            //The lack of an icon will be enough notification
```
Comment (Pacane) [inline sdk/dslink/src/main/java/org/dsa/iot/dslink/DSLinkHandler.java]: Not sure I like this. I'd rather failfast, than assume the icon is there, when it's not.

File: sdk/dslink/src/main/java/org/dsa/iot/dslink/DSLinkHandler.java [hunk 2]
```diff
@@ -91,6 +92,43 @@ public Configuration getConfig() {
...
+        }
+        try {
+            out.close();
+            in.close();
+        } catch (Exception ignore) {
+        }
+        return out.toByteArray();
+    }
+
+
```
Comment (Pacane) [inline sdk/dslink/src/main/java/org/dsa/iot/dslink/DSLinkHandler.java]: formatting

**Comment by a-hansen** [comment]
@Pacane Both issues addressed.  It looks like the build is failing on the integration tests:  

FileSystemException: Cannot copy file to '/home/travis/build/IOT-DSA/sdk-dslink-java/dslink-dart-test/dump-1XF9aJ/dslink-java-etsdb-0.17.0-SNAPSHOT/lib/historian-0.17.0-SNAPSHOT.jar', path = '/home/travis/build/IOT-DSA/sdk-dslink-java/sdk/historian/build/libs/sdk/historian-0.17.0-SNAPSHOT.jar'

**Comment by Pacane** [comment]
@a-hansen right... the integrated-tests only work for the latest version as of right now (the latest SDK is on 0.17.0-snapshot). We can either ignore it and merge or make the tests aware of the different versions.

**Pull request merged**